8.18.5 release notes (#7022)
elastic/security-docs: docs/release-notes/8.18.asciidoc

[[release-notes-header-8.18.0]]
== 8.18

[discrete]
[[release-notes-8.18.5]]
=== 8.18.5

[discrete]
[[enhancements-8.18.5]]
==== Enhancements
* Adds the `detection_rule_upgrade_status` object to snapshot telemetry schema ({kibana-pull}223086[#223086]).
* Reduces {elastic-defend} CPU when processing events from the System process on Windows.
* Reduces {elastic-defend} CPU usage for ETW events, API events, and Behavioral Protections. In some cases, this may be a significant reduction.
* Allows {elastic-defend} to automatically recover in some situations when it loses connectivity with {agent}.
* Shortens the time it takes {elastic-defend} to recover from a `DEGRADED` status caused by communication issues with {agent}.

> * Makes {elastic-defend} malware scan queue operate more efficiently on Windows by not blocking scan requests when an oplock for the file being scanned cannot be acquired.

Reviewer: ```suggestion
* Improves {elastic-defend} malware scan queue efficiency on Windows by not blocking scan requests when an oplock for the file being scanned cannot be acquired.
```

* Due to an issue in macOS, {elastic-defend} would sometimes send network events without `user.name` populated. {elastic-defend} will now identify these events and populate `user.name` if necessary.

[discrete]
[[bug-fixes-8.18.5]]
==== Fixes
* Improves UI copy for the "bulk update with conflicts" modal ({kibana-pull}227803[#227803]).
* Fixes an issue where {elastic-defend} would fail to enable network events on Linux if IPv6 is not supported by the system.

> * Fixes an issue in {elastic-defend} that could result in a crash if a {ls} output configuration is specified containing a certificate that cannot not be parsed.

Reviewer: See same question/comment from https://github.com/elastic/security-docs/pull/7021/files#r2267400469.

[[release-notes-header-8.18.0]]
== 8.18

[discrete]
[[release-notes-8.18.5]]
=== 8.18.5

[discrete]
[[enhancements-8.18.5]]
==== Enhancements
* Adds the `detection_rule_upgrade_status` object to snapshot telemetry schema ({kibana-pull}223086[#223086]).

> * To help identify which parts of `elastic-endpoint.exe` are using a significant amount of CPU, {elastic-defend} on Windows can now include CPU profiling data in diagnostics. To request CPU profiling data using the command line, refer to {fleet-guide}/elastic-agent-cmd-options.html#_options[{agent} command reference]. To request CPU profiling data using {kib}, check the **Collect additional CPU metrics** box when requesting {agent} diagnostics.

Reviewer 2: Unfortunately the CPU profiling feature backport PR wasn't merged into the `8.18` branch until after BC1.  It'll land if they create BC2, but that's unlikely given the short timetable.

```suggestion
* Reduces {elastic-defend} CPU when processing events from the System process on Windows.
* Reduces {elastic-defend} CPU usage for ETW events, API events, and Behavioral Protections. In some cases, this may be a significant reduction.
* Allows the {elastic-defend} to automatically recover in some situations when it loses connectivity with Agent.
* Shortens the time it takes {elastic-defend} to recover from a DEGRADED status caused by communication issues with {elastic-agent}.
* Makes {elastic-defend} malware scan queue operate more efficiently on Windows by not blocking scan requests when an oplock for the file being scanned cannot be acquired.
* Due to an issue in macOS, {elastic-defend} would sometimes send network events without `user.name` populated. {elastic-defend} will now identify these events and populate `user.name` if necessary
```

[discrete]
[[bug-fixes-8.18.5]]
==== Fixes

> * Improves UI copy for the "bulk update with conflicts" modal ({kibana-pull}227803[#227803]).

Author: @nikitaindik We have this one documented in the [8.18.4 RNs](https://www.elastic.co/guide/en/security/8.18/release-notes-header-8.18.0.html#bug-fixes-8.18.4), since the PR originally had the `v8.18.4` label (which was then replaced by `v8.18.5`). Should we remove it from 8.18.4 RNs?

Reviewer 3: Good catch! It seems that it didn't make it into `v8.18.4`. Yeah, let's move it to `v8.18.5` in RNs then.

> * Improves UI copy for the "bulk update with conflicts" modal ({kibana-pull}227803[#227803]).

Reviewer 2: ```suggestion
* Improves UI copy for the "bulk update with conflicts" modal ({kibana-pull}227803[#227803]).
* Fixes an issue where {elastic-defend} would fail to enable network events on Linux if IPv6 is not supported by the system.
* Fixes an issue in {elastic-defend} that could result in a crash if a Logstash output configuration is specified containing a certificate which cannot not be parsed.
```
